### PR TITLE
not sure if I like this...maybe...

### PR DESCRIPTION
**Pull request by demacdonald**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,124 +0,0 @@
...
-
-	// default
-	d: function() {
-		if (meetsLogLevelRequirement('default')) {
-			logUtil.logWithColor(config.colors.default, arguments);
-		}
-	},
-	// success
-	s: function() {
-		if (meetsLogLevelRequirement('success')) {
```
Comment (demacdonald) [inline index.js:35]: this format was tedious... if I ever want to change anything with the functions it is a lot of copy paste... this leads to lots of work, and potential for creating bugs.

File: src/loggingFunctions.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+var logUtil = require('./logUtil');
+var linesUtil = require('./linesUtil');
+var config = require('./config');
+
+module.exports = [
+	//////////////////
+	// main methods //
+	//////////////////
+
+	{	//default
```
Comment (demacdonald) [inline src/loggingFunctions.js]: doing it this way, we can make sure we handle all of our log functions the same.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+'use strict';
```
Comment (demacdonald) [inline src/index.js]: index file looks pretty simple now.

File: src/loggingFunctions.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+var logUtil = require('./logUtil');
```
Comment (demacdonald) [inline src/loggingFunctions.js:1]: this replaces all of our logging functions and builds them all using the same code.Makes changing behavior pretty simple.

**Comment by vajahath** [comment]
Yeah. looks like ⭐️.
I think we need to change `main` property in `package.json` file.
OR
keep the entry point `index.js` in the root. 👈 this may be good for better readability. Keeping the gateway  right in the root looks nicer than keeping it somewhere else.

Can we merge this to the [experimental](https://github.com/vajahath/lme/tree/experimental) so that we can develop on the top of this?

**Comment by demacdonald** [comment]
Can merge into my log level branch and then merge that in whenever  you'd like.

**Comment by demacdonald** [comment]
Only saying that because I built this off my log level branch

**Pull request closed without merging**